### PR TITLE
use the `required` instruction to determine if the environment fits the sample code's required.

### PR DESCRIPTION
**Pull request by wadefelix**
### PR types
Function optimization

### PR changes
Others

### Describe
1 put a instruction `# required: gpu`(for example) in the sample code
2 this piece of code will only run in the GPU-equipped CI pipelines, and be omitted in other pipelines.
3 the CI pipelines can specify its capacity by shell environment variable `SAMPLE_CODE_TEST_CAPACITY`

see [2.1 文档改版方案](http://agroup.baidu.com/paddlepaddle/md/article/3821189) for detail.

### CODE REVIEW

File: tools/sampcd_processor.py [hunk 1]
```diff
@@ -117,122 +320,108 @@ def sampcd_extract_to_file(srccom, name, htype="def", hname=""):
...
+        if matched == True:
+            tfname = os.path.join(SAMPLECODE_TEMPDIR, '{}_example{}'.format(
+                name, '.py'
+                if len(codeblocks) == 1 else '_{}.py'.format(y + 1)))
+            with open(tfname, 'w') as tempf:
+                sampcd = insert_codes_into_codeblock(cb, name)
+                tempf.write(sampcd)
+            sample_code_filenames.append(tfname)
+        else:
+            if matched is None:
```
Comment (MingMingShangTian) [inline tools/sampcd_processor.py]: 这里的if 可以和上边的else 合并为 elif， 减少一层缩进

**Comment by wadefelix** [comment]
`import math` should be removed.

File: tools/sampcd_processor.py (hunk 1 above)
Comment (wadefelix) [reply tools/sampcd_processor.py]: done

**Review by MingMingShangTian — APPROVED** [review]
LGTM

**Pull request merged**